x/distribution/types/delegator_info.go

```diff
@@ -42,16 +50,49 @@ func (di DelegationDistInfo) WithdrawRewards(wc WithdrawContext, vi ValidatorDis
 	vi = vi.UpdateTotalDelAccum(wc.Height, totalDelShares)
 
 	if vi.DelAccum.Accum.IsZero() {
+		// NOTE: under the situation which a delegation was created with
+		//       zero shares, this next line of code would ensure
+		//       that no delegation-accum invariance relative to the total
+		//       delegation accum kept by validator_info. Ideally however
+		//       delegations with zero shares should never be created.
+		di.DelPoolWithdrawalHeight = wc.Height
 		return di, vi, fp, DecCoins{}
 	}
 
 	vi, fp = vi.TakeFeePoolRewards(wc)
 
 	accum := di.GetDelAccum(wc.Height, delegatorShares)
 	di.DelPoolWithdrawalHeight = wc.Height
-	withdrawalTokens := vi.DelPool.MulDec(accum).QuoDec(vi.DelAccum.Accum)
+
+	var withdrawalTokens DecCoins
+	if accum.Equal(vi.DelAccum.Accum) {
+		// required due to rounding faults
```

Comment:
ok, this should only happen for a validator with a single delegator, right?

Reply:
correct